R/derive_merged.R

```diff
@@ -173,136 +180,188 @@
+#'   dataset_add = vs,
+#'   by_vars = exprs(STUDYID, USUBJID),
+#'   order = exprs(convert_dtc_to_dtm(VSDTC)),
+#'   mode = "last",
+#'   new_vars = exprs(LSTWT = VSSTRESN),
+#'   filter_add = VSTESTCD == "WEIGHT"
+#' ) %>%
+#'   select(USUBJID, AGE, AGEU, LSTWT)
+#'
+#' @caption Handling duplicates (`check_type`)
+#' @info The source records are checked regarding duplicates with respect to the
+#'   by variables and the order specified.
+#'   By default, a warning is issued if any duplicates are found.
+#'   Note the results here with a new vital signs dataset containing a
+#'   duplicate last weight assessment date.
+#' @code [expected_cnds = "duplicate_records"]
+#' vs_dup <- tribble(
+#'   ~DOMAIN,  ~USUBJID, ~VSTESTCD, ~VISIT,      ~VSSTRESN, ~VSDTC,
+#'   "VS",     "01",     "WEIGHT",  "WEEK 2",        81.1, "2013-09-24",
+#'   "VS",     "01",     "WEIGHT",  "WEEK 4",        82.6, "2013-09-24"
+#' ) %>%
+#'   mutate(STUDYID = "AB42")
 #'
-#' # Merge last weight to adsl
-#' adsl <- tribble(
-#'   ~STUDYID,   ~USUBJID, ~AGE,   ~AGEU,
-#'   "PILOT01", "01-1302",   61, "YEARS",
-#'   "PILOT01", "17-1344",   64, "YEARS"
-#' )
+#' derive_vars_merged(
+#'   dm,
+#'   dataset_add = vs_dup,
+#'   by_vars = exprs(STUDYID, USUBJID),
+#'   order = exprs(convert_dtc_to_dtm(VSDTC)),
+#'   mode = "last",
+#'   new_vars = exprs(LSTWT = VSSTRESN),
+#'   filter_add = VSTESTCD == "WEIGHT"
+#' ) %>%
+#'   select(USUBJID, AGE, AGEU, LSTWT)
+#'
+#' @info For investigating the issue, the dataset of the duplicate source records
+#'   can be obtained by calling `get_duplicates_dataset()`:
+#' @code
+#' get_duplicates_dataset()
+#'
+#' @info Common options to solve the issue:
+#' - Restricting the source records by specifying/updating the `filter_add` argument.
+#' - Specifying additional variables for `order` - this is the most common approach
+#'   adding something like a sequence variable.
+#' - Setting `check_type = "none"` to ignore any duplicates, but then in this case
+#'   the last occurring record would be chosen according to the sort order of the
+#'   input `dataset_add`.
+#'
+#' @caption Modify values dependent on the merge (`new_vars` and `missing_values`)
+#' @info For the last occurring weight for each subject, add a categorisation of
+#'   which visit it occurred at to the demographics dataset.
+#'
+#' - In the `new_vars` argument, other functions can be utilised to modify the
+#'   merged values. For example, in the below case we want to categorise the
+#'   visit as `"BASELINE"` or `"POST-BASELINE"` using `if_else()`.
+#' - The `missing_values` argument assigns a specific value for subjects with
+#'   no matching observations - see subject `"03"` in the below example.
+#' @code
+#' derive_vars_merged(
+#'   dm,
+#'   dataset_add = vs,
+#'   by_vars = exprs(STUDYID, USUBJID),
+#'   order = exprs(convert_dtc_to_dtm(VSDTC)),
+#'   mode = "last",
+#'   new_vars = exprs(
+#'     LSTWTCAT = if_else(VISIT == "BASELINE", "BASELINE", "POST-BASELINE")
```

Comment:
As we are building on the previous example, I'd add LSTWT here too for clarity. I know you talk about creating more than one variable later though, so it's up to you.

Reply:
yes think i'll leave this one as then it keeps the creating multiple variables bit separate later, and these 2 examples are easy enough to follow on from one another